certs/secrets.go

```diff
@@ -0,0 +1,73 @@
+package certs
+
+import (
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+// NewSecretsFromCertificates returns a list of new secrets, 1 for each certificate
+func NewSecretsFromCertificates(input *Certificates) []*corev1.Secret {
```

Comment:
Do you want to check for each input keypair and return an error if any is missing (like NewCertificatesFromSecrets)?